daceml/transformation/reshape_elimination.py

```diff
@@ -0,0 +1,146 @@
+import functools
+from collections import deque
+from typing import Dict
+
+import dace
+from dace import registry, properties, subsets
+from dace.sdfg import nodes, utils as sdfg_utils
+from dace.transformation import transformation as xf
+
+import daceml.onnx as donnx
+from daceml.util import utils
+
+
+def expand_library_nodes_except_reshape(self, recursive=True):
+    states = list(self.states())
+    while len(states) > 0:
+        state = states.pop()
+        expanded_something = False
+        for node in list(state.nodes()):  # Make sure we have a copy
+            if isinstance(node, nodes.NestedSDFG):
+                node.sdfg.expand_library_nodes()  # Call recursively
+            elif isinstance(node, nodes.LibraryNode) and not isinstance(
+                    node, donnx.ONNXReshape):
+                impl_name = node.expand(self, state)
+                print(
+                    "Automatically expanded library node \"{}\" with implementation \"{}\"."
+                    .format(str(node), impl_name))
+                # We made a copy of the original list of nodes, so we keep
+                # iterating even though this list has now changed
+                if recursive:
+                    expanded_something = True
+        if expanded_something:
+            states.append(state)  # Nodes have changed. Check state again
+
+
+@registry.autoregister_params(singlestate=True)
+@properties.make_properties
+class ReshapeElimination(xf.Transformation):
+    """ Merge a reshape into a preceding or following nested SDFG call.
+    """
+    # pattern matching only checks that the type of the node matches,
+    _reshape_node = xf.PatternNode(donnx.ONNXReshape)
+    _access_node = xf.PatternNode(nodes.AccessNode)
+    _nsdfg = xf.PatternNode(nodes.NestedSDFG)
+
+    @staticmethod
+    def expressions():
+        return [
+            sdfg_utils.node_path_graph(ReshapeElimination._reshape_node,
```

Comment:
oh, now I'm confused. I thought it was nsdfg->reshape->access or access->reshape->nsdfg?

Reply:
This is elimination of an `ONNXReshape`, not an Access node`